trunk/SUAVE/Methods/Noise/Fidelity_One/Propeller/compute_broadband_noise.py

```diff
@@ -0,0 +1,111 @@
+    R_tip          = propeller.tip_radius                                   
+    beta           = propeller.twist_distribution                           # twist distribution  
+    t              = propeller.max_thickness_distribution                   # thickness distribution 
+                                                                            
+    n              = len(R)                                                 
+    S              = np.sqrt(x**2 + y**2 + z**2)                            # distance between rotor and the observer   
+    V_tip          = R_tip*omega                                            # blade_tip_speed   
+    V_07           = V_tip*0.70/(Units.feet)                                # blade velocity at r/R_tip = 0.7 
+    St             = 0.28                                                   # Strouhal number             
+    t_avg          = np.mean(t)/(Units.feet)                                # thickness
+    c_avg          = np.mean(c)/(Units.feet)                                # average chord  
+    beta_07        = beta[round(n*0.70)]                                    # blade angle of attack at r/R = 0.7
+    h_val          = t_avg*np.cos(beta_07) + c_avg*np.sin(beta_07)          # projected blade thickness                   
+    f_peak         = (V_07*St)/h_val                                        # V - blade velocity at a radial location of 0.7              
+    A_blade        = (np.trapz(c, x = R))/(Units.feet**2)                    # area of blade      
+    CL_07          = 2*np.pi*beta_07
+    S_feet         = S/(Units.feet)
+    
+    SPL_300ft      = np.atleast_2d(10*np.log10(((6.1E-27)*A_blade*V_07**6)/(10**-16)) + 20*np.log(CL_07/0.4)) 
+    SPL_300ft      = np.repeat(np.repeat(SPL_300ft.T,num_mic, axis = 1)[:,:,np.newaxis],num_prop, axis = 2)
+    SPL_vals       = SPL_300ft - 20*np.log10(S_feet/300)                       
+    SPL_vals       = SPL_300ft - 20*np.log10(S_feet/300)                       
+    
+    # estimation of A-Weighting for Vortex Noise  
+    f_v            = np.array([0.5*f_peak,1*f_peak,2*f_peak,4*f_peak,8*f_peak,16*f_peak]).T # spectrum
+    fr             = np.array([0.5,1,2,4,8,16])                                           # frequency ratio  
+    weights        = np.atleast_2d(np.array([7.92 , 4.17 , 8.33 , 8.75 ,12.92 , 13.33]))
+    SPL_weight     = np.repeat(np.repeat(np.repeat(weights, num_prop, axis = 0)[np.newaxis,:,:], num_mic, axis = 0)[np.newaxis,:,:,:], num_cpt, axis = 0)    # SPL weight
+    SPL_v          = np.repeat(SPL_vals[:,:,:,np.newaxis], 6 , axis = 3) - SPL_weight                          # SPL correction 
+    dim            = len(fr)
+    C              = np.zeros((num_cpt,num_mic,num_prop,dim))
+    p_pref_bb_dBA  = np.zeros((num_cpt,num_mic,num_prop,dim-1))
+    SPL_bb_dbAi    = np.zeros((num_cpt,num_mic,num_prop,dim))
+    
+    for i in range(num_cpt): 
```

Comment:
vectorize

Reply:
I can vectorize it any more because of the type of frequency spectrum the noise is in. Additionally, the A-weighting function is already as vectorized as it can be.

Reply:
Note that this will change sometime this year when I implement new methods